subworkflows/local/sourmash_profiling.nf

```diff
@@ -25,16 +43,42 @@ workflow SOURMASH_PROFILING {
                 meta.id = "k31"
                 [ meta, signatures ]
         }
+
     SOURMASH_COMPARE(ch_compare, seqtype, [], true, true)
     ch_compare_matrix = SOURMASH_COMPARE.out.matrix
     ch_compare_csv = SOURMASH_COMPARE.out.csv
     ch_versions = ch_versions.mix(SOURMASH_COMPARE.out.versions)
 
     // gather against database
+    // prep all combinations of input files with combined databases
+    ch_input_gather = ch_signatures
```

Comment:
Is there a reason to do this and not provide the list of paths as another separate variable? Perhaps I don't understand the final output shape of `ch_input_gather`, could you add a comment about it?

Also in the underlying `SOURMASH_GATHER`, you seem to refer to `databases` as a single database path. Is that the case? Or is it a list of paths and if so does it stage all of them correctly?

Reply:
It is a list of paths, so when you give it multiple database paths in the CSV it will run them all together once per sample, and currently how I have it set up it stages it correctly yes. 
So if you have 2 databases and 2 samples, there should only be 2 ch_input_gather feeds, one for each sample so that it runs all the databases once for each sample.

Reply:
and in sourmash gather it says `path (databases)` but it gets passed as a list and it stages correctly, I went into the work directory for this process and checked them